pandas/formats/format.py

```diff
@@ -906,8 +927,59 @@ def get_col_type(dtype):
                          if x else '{}') for x in row]
             else:
                 crow = [x if x else '{}' for x in row]
+            if i < clevels and self.fmt.header and self.multicolumn:
+                # sum up columns to multicolumns
+                row2 = list(crow[:ilevels])
+                ncol = 1
+                coltext = ''
+
+                def append_col():
+                    # write multicolumn if needed
+                    if ncol > 1:
+                        row2.append('\\multicolumn{{{0:d}}}{{{1:s}}}{{{2:s}}}'
+                                    .format(ncol, multicolumn_format,
+                                            coltext.strip()))
+                    # don't modify where not needed
+                    else:
+                        row2.append(coltext)
+                for c in crow[ilevels:]:
+                    if c.strip():  # if next col has text, write the previous
+                        if coltext:
+                            append_col()
+                        coltext = c
+                        ncol = 1
+                    else:  # if not, add it to the previous multicolumn
+                        ncol += 1
+                if coltext:  # write last column name
+                    append_col()
+                crow = row2
+            if i >= nlevels and self.fmt.index and self.multirow:
+                # perform look-ahead to determine which rows can be joined
+                for j in range(ilevels):
+                    if crow[j].strip():
+                        nrow = 1
+                        for r in strrows[i + 1:]:
+                            if not r[j].strip():
+                                nrow += 1
+                            else:
+                                break
+                        if nrow > 1:
+                            # overwrite non-multirow entry
+                            crow[j] = '\\multirow{{{0:d}}}{{*}}{{{1:s}}}'\
+                                      .format(nrow, crow[j].strip())
+                            # save when to end the current block with \cline
+                            clinebuf.append([nrow, j + 1])
             buf.write(' & '.join(crow))
             buf.write(' \\\\\n')
+            if self.multirow and i < len(strrows) - 1:
+                # during main block, check if sub-block transition takes place
+                # print \cline to distinguish \multirow-areas
+                for cl in clinebuf:
+                    cl[0] -= 1
+                    if cl[0] == 0:
+                        buf.write('\cline{{{0:d}-{1:d}}}\n'.format(cl[1],
```

Comment:
In the above block, i iterate over the index levels and check if a multirow should be used. If i find a multirow, i save the number of rows it spans, and it's level (indented by 1) in clinebuf (list, because multiple multirows can occur at the same time). When writing this line and the following, i decrease the first entry in the buffered pairs. If this value reaches zero, it marks the point where the multirow ends, and i place the cline. When all levels have been checked, i flush the buffer.